WordPress/Classes/ViewRelated/Domains/Views/DomainResultView.swift

```diff
@@ -0,0 +1,114 @@
+import SwiftUI
+
+struct DomainResultView: View {
+
+    @State private var subtitleHeight: CGFloat = .zero
+    let domain: String
+    let dismiss: () -> Void
+
+    private struct AttributedLabel: UIViewRepresentable {
+        typealias UIViewType = UILabel
+        @Binding var dynamicHeight: CGFloat
+
+        func makeUIView(context: Context) -> UILabel {
+            let label = UIViewType()
+            label.numberOfLines = 0
+            label.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+            label.lineBreakMode = .byWordWrapping
+            return label
+        }
+
+        func updateUIView(_ uiView: UILabel, context: Context) {
+            configuration(uiView)
+
+            // Makes the label vertically hug its multi-line content
+            DispatchQueue.main.async {
+                dynamicHeight = uiView.sizeThatFits(CGSize(width: uiView.bounds.width, height: CGFloat.greatestFiniteMagnitude)).height
+            }
+        }
+
+        var configuration: (UIViewType) -> Void
+    }
+
+    var body: some View {
```

Comment:
Changing orientation makes the site name disappear. I suppose it can be fixed with the same solution to fix attributed string + dynamic type, by observing trait collection changes

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/d1630588-669a-4786-916d-7c609f7fd0da

Reply:
Thanks for the suggestion! 
I can listen to trait changes in SwiftUI through code such as:
```
@Environment(\.sizeCategory) var sizeCategory
```
but I'm not sure how to refresh the view based on the size category change.

So I tried a different approach using `GeometryReader` to get the current screen width and use that in the `sizeThatFits` (instead of using the label's bounds like I did before). This works somewhat because the label now is sized correctly based on the current dynamic type size and device orientation. However, it doesn't respond to changes to dynamic type size after this screen is already displayed.

I looked through `StatsTotalInsightsCell` for inspiration but the missing piece for me is how to make the size category change re-draw the screen.